source/MRMesh/MRMeshLoadObj.cpp

```diff
@@ -694,11 +696,41 @@ tl::expected<std::vector<NamedMesh>, std::string> fromSceneObjFile( const char*
 
                     assert( vts.size() == vs.size() );
                     for ( int j = 0; j < vs.size(); ++j )
-                        uvCoords[VertId{ vs[j] - 1 }] = textureVertices[ vts[j] - 1 ];
+                    {
+                        if ( texCoords[vs[j] - 1] == vts[j] - 1 )
+                            continue;
+
+                        const std::lock_guard<std::shared_mutex> lock( mutex );
```

Comment:
Better to lock the mutex once for the whole loop rather than on every iteration.